plugin_tests/dataone_register_test.py

```diff
@@ -0,0 +1,320 @@
+import pytest
+from tests import base
+from girder.api.rest import RestException
+
+'''Tests for the methods in dataone_register.py. Some of these tests use live requests, while others use
+mocked JSON responses/data structures. '''
+
+
+def setUpModule():
+
+    base.enabledPlugins.append('wholetale')
+    base.startServer()
+
+
+def tearDownModule():
+
+    base.stopServer()
+
+
+class TestDataONERegister(base.TestCase):
+
+    def test_find_initial_pid(self):
+        from server.dataone_register import find_initial_pid
+
+        # Test that the regex is working for search.dataone urls
+        pid = 'https://search.dataone.org/#view/urn:uuid:7ec733c4-aa63-405a-a58d-1d773a9025a9'
+        res = find_initial_pid(pid)
+        assert res == 'urn:uuid:7ec733c4-aa63-405a-a58d-1d773a9025a9'
+
+        # Test that the regex is working for paths coming from the coordinating node
+        pid = 'https://cn.dataone.org/cn/v2/object/urn:uuid:6f5533ab-6508-4ac7-82a3-1df88ed4580e'
+        res = find_initial_pid(pid)
+        assert res == pid
+
+        # Test that if nothing was found, None is returned
+        bad_url = 'has2:"//badurl.bxwcom/a_bad_/url'
+        res = find_initial_pid(bad_url)
+        assert res == bad_url
+
+    def test_find_resource_pid(self):
+        from server.dataone_register import find_resource_pid
+
+        # Test the case where no data object could be located
+        with pytest.raises(RestException) as error:
+            bad_url = 'https:"//badurl.com/a_bad_/url'
+            find_resource_pid(bad_url)
+
+    def test_get_package_files_metadata(self):
+        ''' Test that the metadata in a package is getting added to the list of files'''
+        from server.dataone_register import get_package_files
+
+        full_meta_data = [{'identifier': 'urn:uuid:f438a8d5-7965-4ca2-aad6-88f694e5afe5', 'fileName': 'iso19139.xml',
+        'formatId': 'http://www.isotc211.org/2005/gmd', 'formatType': 'METADATA', 'size': 14735,
+        'title': 'Collaborative Research: A Synthesis of Existing and New Observations of Air-Snowpack Exchanges to'
+                           ' Assess the Arctic',
+        'documents': ['urn:uuid:f438a8d5-7965-4ca2-aad6-88f694e5afe5',
+                   'resource_map_urn:uuid:23acf3bd-27c7-4736-a6ef-c141698d66b8',
+                   'resource_map_urn:uuid:1570fb91-69e8-4c1d-9b9a-292edb5820af']}]
+
+        data = []
+
+        expected_result = {'iso19139.xml': {'size': 14735}}
+
+        # In this case, the primary metadata object is the same as the metadata object
+        fileList = get_package_files(data, full_meta_data, full_meta_data)
+        assert fileList == expected_result
```

Comment:
Could you use `self.assertDictEqual` for `dict` comparisons? It yields much nicer tracebacks than raw `assert` in case something goes awry.